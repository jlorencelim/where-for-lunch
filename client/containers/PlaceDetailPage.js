import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import placeActions from '../actions/placeActions';
import Place from '../components/Place/Place';


class PlaceDetailPage extends Component {
  static propTypes = {
    place: PropTypes.object,
    fetchPlaceDetail: PropTypes.func,
    match: PropTypes.object,
  };

  componentDidMount() {
    const { fetchPlaceDetail, match: { params: { id } } } = this.props;

    fetchPlaceDetail(id);
  }

  render() {
    const { place } = this.props;

    return (
      <Place place={place} detailedData />
    );
  }
}

const mapStateToProps = state => ({
  place: state.place,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  fetchPlaceDetail: placeActions.fetchPlaceDetail,
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PlaceDetailPage);
