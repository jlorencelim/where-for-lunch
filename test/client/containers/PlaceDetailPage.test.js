import React from 'react';
import { createMockStore } from 'redux-test-utils';
import PlaceDetailPage from 'client/containers/PlaceDetailPage';
import { shallowWithStore } from '../../utils';

test('With Jest snapshot, PlaceDetailPage container renders successfully', () => {
  const state = {
    place: {
      id: 1,
      photos: ['photo.png'],
      rating: 3.5,
    },
  };

  const store = createMockStore(state);
  const wrapper = shallowWithStore(<PlaceDetailPage />, store);

  expect(wrapper).toMatchSnapshot();
});
