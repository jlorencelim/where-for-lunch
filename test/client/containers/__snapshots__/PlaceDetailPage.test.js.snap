// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, PlaceDetailPage container renders successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(PlaceDetailPage) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "fetchPlaceDetail": [Function],
      "place": Object {
        "id": 1,
        "photos": Array [
          "photo.png",
        ],
        "rating": 3.5,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "fetchPlaceDetail": [Function],
        "place": Object {
          "id": 1,
          "photos": Array [
            "photo.png",
          ],
          "rating": 3.5,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
